recipes/agent.rb

```diff
@@ -31,6 +31,8 @@
 end
 
 service 'blackfire-agent' do
+  not_if { node[cookbook_name]['agent']['server_id'].empty? }
+  not_if { node[cookbook_name]['agent']['server_token'].empty? }
```

Comment:
one last thing: you also need to check for `nil?` before `empty?` as defaults are `nil` and not empty strings.